pkg/accumulator/resaccumulator.go

```diff
@@ -64,6 +64,15 @@ func (ra *ResAccumulator) GetTransformerConfig() *config.TransformerConfig {
 
 func (ra *ResAccumulator) MergeVars(incoming []types.Var) error {
 	for _, v := range incoming {
+		// Warning: Do not change GvknEquals to Equals until the
```

Comment:
If `namespace` is part of var declaration, does it require every var to specify a namespace?
If the resource has a namespace but a var specified to match it doesn't have the namespace, does this fail? I would expect this to pass.

Reply:
@Liujingfang1 I'm trying to solve the issue in the following PR.  This PR was just to show the issue. 

The main complexity in solving the issue:
- namespacetransformer changing the namespace of the resources. I think the way the mergevars and resolvevar work should be fine.
- fundamentally "no namespace" in the var means "any namespace" where in ResId.Equals understand "no namespace" as "default" if it is not a cluster wide resources. We are closer from what you coded in the PatchTarget and Selector code.

Still working on the solution (in the next PR).

Reply:
sounds good.

Reply:
done. Added comment on preserving backward compatibility.